cpp/dolfinx/io/XDMFFile.cpp

```diff
@@ -350,6 +351,21 @@ XDMFFile::read_meshtags(const mesh::Mesh<double>& mesh, std::string name,
 
   pugi::xml_node values_data_node
       = grid_node.child("Attribute").child("DataItem");
+  if (!attribute_name.empty())
```

Comment:
Use `std::optional`, in which case this would become `if (attribute_name)`.

Reply:
Done.